### ISSUE TITLE
Gitlab Runners in UDS.is hitting dockerhub anonymous pull rate limit

### ISSUE THREAD

**Issue by joelmccoy** [defenseunicorns/uds-software-factory#130]
Example run: https://gitlab.uds.is/defenseunicorns/uds/packages/uds-package-meilisearch/-/jobs/1647#L132

It appears we are hitting the rate limit for dockerhub anonymous pulls (100 per 6 hours).  Most of our test pipelines are pulling images from dockerhub and we quickly hit this rate limit.

Some options to look into:
- Adding a dockerhub robot account creds to the setup scripts
- Utilize a [dependency proxy](https://docs.gitlab.com/ee/development/packages/dependency_proxy.html)

**Comment by zachariahmiller** [comment]
Yeah this is a frustrating problem. The dependency proxy is definitely a possible option with its own set of potential implementation issues and annoyances. 

I'm not at all opposed to adding dockerhub robot creds and thats probably the path of least resistance technically speaking. This is something that we would need to work through as an organization though in terms of having an enterprise docker account and obtaining said credentials. I'm not sure what the appetite would be for this. 

 Its also unclear to me how pressing of an issue this is to solve at the present moment given some of the changes around where development is actually occurring.
